lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -115,3 +119,51 @@ func (c *ImageCache) GetImages() []*metadata.ImageConfig {
 
 	return result
 }
+
+func (c *ImageCache) GetImageByDigest(digest digest.Digest) *metadata.ImageConfig {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	var imageID string
+
+	for _, image := range c.cache {
+		if strings.Contains(image.ImageID, "sha") {
+			imageID = image.ImageID
+		} else {
+			imageID = "sha256:" + image.ImageID
+		}
+
+		if imageID == string(digest) {
+			newImage := new(metadata.ImageConfig)
+			*newImage = *image
+			return newImage
+		}
+	}
+
+	return nil
+}
+
+// Looks up image by reference.Named
+func (c *ImageCache) GetImageByNamed(named reference.Named) *metadata.ImageConfig {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	for _, image := range c.cache {
+		ref := reference.WithDefaultTag(named)
+		if namedtag, ok := ref.(reference.NamedTagged); ok {
+			if named.RemoteName() == image.Name && namedtag.Tag() == image.Tag {
+				newImage := new(metadata.ImageConfig)
+				*newImage = *image
+				return newImage
+			}
+		} else {
+			if named.Name() == image.ImageID {
```

Comment:
Is this where we would implement lookup by short ID? If so is it viable to do so now:
1. use strings.HasPrefix instead of ==
2. continue iterating after the first match and abort with error if there's a second match